### PR TITLE
Update/dashboard redux

### PR DESCRIPTION
**Pull request by paulmatchett**
redux store implemented into dashboard

### CODE REVIEW

**Comment by William-Unosquare** [comment]
I'm going to work on this a bit on Monday.

We shouldn't merge in two different styles of reducer so I'm going to update it to the style already used. I'll also add some selectors in.

**Comment by William-Unosquare** [comment]
I looked at this more closely this morning. There's a lot that needs to be changed. I'm going to say we don't merge this in right now and Kirsty had the same opinion. 

I was hoping I'd get it fixed up this morning but the reason I'm not going to do that is that I'm not sure if there are reasons behind some of the stuff @paul-matchett has done here. And there really is a lot to change. I'd much rather pair up with Paul and go through it with him.

I think the main area that needs to be worked on here is the actions & action creators. There seems to be an extra layer added into all this logic that the action creators should be handling. For example you have all this in the actions file: 

``` javascript
export const showBookingModal = () => {
  return {
    type: actionTypes.SHOW_BOOKING_MODAL,
    bookingModalOpen: true,
  };
};

 export const hideBookingModal = () => {
  return {
    type: actionTypes.HIDE_BOOKING_MODAL,
    bookingModalOpen: false,
  };
};
```

``` javascript
export const toggleBookingModal = bookingModalOpen => {
  return dispatch => {
    if (bookingModalOpen) {
      dispatch(showBookingModal());
    } else {
      dispatch(hideBookingModal());
    }
  };
};
```

When this would be enough:

``` javascript
 export const toggleBookingModal = shouldDisplay => {
  return {
    type: TOGGLE_BOOKING_MODAL,
    payload: shouldDisplay,
  };
};
```

You should then dispatch `toggleBookingModal` elsewhere in the code, where you'd be toggling the modal.

**Review by William-Unosquare — CHANGES_REQUESTED** [review]
See above ^

**Review by William-Unosquare — APPROVED** [review]

**Comment by William-Unosquare** [comment]
This is hopefully all good to merge now after refactoring, need someone else to review

File: web-app/src/actions/dashboard.js
Comment (kirsty-unosquare) [inline web-app/src/actions/dashboard.js:26]: Not keen on the payload name `open`. If it's toggle then it could also be closing the modal. Maybe name it `toggle` or something similar. :)

File: web-app/src/components/BookingModal/BookingModalForm/container.js
Comment (kirsty-unosquare) [inline web-app/src/components/BookingModal/BookingModalForm/container.js]: Just my own thought. But seeing a lot of `this.props...`. I think it would be neater to do ` const { booking: { halfDay, start, eventType }} = this.props;`

File: web-app/src/components/BookingModal/container.js
Comment (kirsty-unosquare) [inline web-app/src/components/BookingModal/container.js:59]: Same here. `open` doesn't make sense when your toggling a modal. In this case your closing it.

File: web-app/src/components/Legend/container.js
Comment (kirsty-unosquare) [inline web-app/src/components/Legend/container.js]: Just a heads up react now declare `componentWillUpdate` as unsafe. From the top of my head I think they prefer to use `componentDidUpdate`.
https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate

File: web-app/src/pages/Dashboard/container.js
Comment (kirsty-unosquare) [inline web-app/src/pages/Dashboard/container.js]: Same here

**Review by kirsty-unosquare — COMMENTED** [review]
Had a look through, just a couple of tidy ups. A lot to look over so wouldn't mind someone else reviewing it.

File: web-app/src/components/BookingModal/container.js
Comment (William-Unosquare) [reply web-app/src/components/BookingModal/container.js:59]: To me, it makes more sense when considered as a bool. `open: true`, `open: false`.  

`toggle: true`, `toggle: false` sounds less descriptive.  Lemme know what you think.

**Pull request merged**
